lib/ansible/modules/cloud/amazon/ec2_transit_gateway.py

```diff
@@ -439,7 +439,7 @@ def ensure_tags(self, tgw_id, tags, purge_tags):
         if to_update:
             try:
                 if not self._check_mode:
-                    AWSRetry.exponential_backoff()(self._connection.create_tags)(
+                    AWSRetry.exponential_backoff(catch_extra_error_codes=['InvalidTransitGatewayID.NotFound'])(self._connection.create_tags)(
```

Comment:
A waiter would be nicer in this module after resource creation instead. There wasn't a built-in one though.

Reply:
Oh, there's already an option to set `wait: True` for this module, which will avoid this possible exception. I'm removing this module's changes.